### ISSUE TITLE
TestSyncCheckpointEarlierThanHead is flaky

### ISSUE THREAD

**Issue by Kubuxu** [filecoin-project/lotus#4716]
```
    github.com/filecoin-project/lotus/chain/exchange.(*client).doRequest
        /home/circleci/project/chain/exchange/client.go:143
    sync_test.go:335: 
        	Error Trace:	sync_test.go:335
        	            				sync_test.go:831
        	Error:      	Should be true
        	Test:       	TestSyncCheckpointEarlierThanHead
```

**Comment by dirkmc** [comment]
Also TestSyncCheckpointHead:
https://app.circleci.com/pipelines/github/filecoin-project/lotus/11547/workflows/1adad4f2-7b1a-4d6b-8578-556c4ca65d0a/jobs/91695

**Comment by Stebalien** [comment]
Did we actually try to figure out what was causing this? If so, can we write that up here?

**Comment by Stebalien** [comment]
Nevermind, the bug was here:

https://github.com/filecoin-project/lotus/blob/4688da51781a1f3249b534e962f8824f323607e3/chain/sync_test.go#L350-L355

We need to break when err is nil (i.e., we _have_ the tipset), not when it's non-nil (we don't have the tipset. The extra "allow for syncing and validation sleep below" was hiding the issue.

Fixed in #6107.
